### PR TITLE
feature DocSearch V3

### PR DESCRIPTION
**Pull request by natacha-h**
| Q             | A
| ------------- | ---
| Branch?       | feature-DocSearch-v3
| Tickets       | [ #325 ](https://github.com/api-platform/website/issues/325)
| License       | MIT

Used DocSearch v3 React component (installed as dependency), modified css in new file _docsearchcustom.scss, added appID in .env.gatsby

### CODE REVIEW

File: .env.gatsby [hunk 1]
```diff
@@ -1,6 +1,7 @@
-GATSBY_DOCSEARCH_API_KEY = 3ec989b752d176d177da4cfe814eee11
```
[the remark this answers is not in the document]
Comment (natacha-h) [reply .env.gatsby:1]: I think it did, using the old one throws an error (403 "invalid api-key or app-id"). And there was a new one when I checked on Algolia so I used the new one 🤷🏼

File: src/components/layout/Search.js [hunk 2]
```diff
@@ -1,87 +1,33 @@
...
+  return (
+    <div className={classNames('search', className)}>
+      <DocSearch
+        appId={process.env.GATSBY_DOCSEARCH_APP_ID}
+        apiKey={process.env.GATSBY_DOCSEARCH_API_KEY}
+        indexName={process.env.GATSBY_DOCSEARCH_INDEX_NAME}
+        searchParameters={{
+          facetFilters: [`version:v${currentVersion}`],
+        }}
+        placeholder="SEARCH..."
```
[the remark this answers is not in the document]
Comment (natacha-h) [reply src/components/layout/Search.js:18]: Oh, yeah, I forgot I should do this 😅

**Review by dunglas — APPROVED** [review]

**Comment by Fabious** [comment]
LGTM

File: src/components/layout/Search.js [hunk 3]
```diff
@@ -1,87 +1,32 @@
...
-      </div>
-    );
-  }
-}
+import { DocSearch } from '@docsearch/react';
+import { currentVersion } from '../../../constants';
+import '@docsearch/css';
+
+const Search = ({ className }) => {
+  return (
```
Comment (alanpoulain) [inline src/components/layout/Search.js:9]:
```suggestion
const Search = ({ className }) => (
```

**Review by alanpoulain — APPROVED** [review]

File: src/styles/vendor/_docsearchcustom.scss [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    }
+
+    .DocSearch-Title {
+      font-size: $text-small;
+      color: var(--title);
+    }
+    
+    @include mq($max-width: $vw-medium) {
+      min-width: 450px!important;
+      max-width: 450px!important;
```
Comment (Fabious) [inline src/styles/vendor/_docsearchcustom.scss:125]:
```suggestion
      min-width: 450px !important;
      max-width: 450px !important;
```

**Pull request merged**